code/modules/power/fuel_assembly/fuel_assembly.dm

```diff
@@ -0,0 +1,94 @@
+/obj/item/fuel_assembly
+	name = "fuel rod assembly"
+	icon = 'icons/obj/machines/power/fusion.dmi'
+	icon_state = "fuel_assembly"
+	layer = 4
+
+	var/material_name
+	var/percent_depleted = 1
+	var/radioactivity = 0
+	var/luminescence = 0
+	var/initial_amount
+
+/obj/item/fuel_assembly/Initialize(mapload, var/list/makeup, var/_material, var/_colour)
```

Comment:
Please avoid reordering Initialize() arguments for /obj stuff, stacks are a huge pain already due to having amount before material type.